rainier-core/src/main/scala/com/stripe/rainier/core/Distribution.scala

```diff
@@ -5,46 +5,17 @@ import com.stripe.rainier.compute.Real
 /**
   * Basic probability distribution trait
   */
-trait Distribution[T] extends Likelihood[T] { self =>
-  def logDensity(t: T): Real
-  def logDensities(list: Seq[T]): Real = Real.sum(list.map(logDensity))
+abstract class Distribution[T, P](implicit placeholder: Placeholder[T, P])
+    extends Likelihood[T] {
+  def logDensity(t: P): Real
+  def logValueDensity(t: T): Real = logDensity(placeholder.wrap(t))
```

Comment:
This name makes it sound to me like we're calculating the density of a log value (or at least its ambiguous). What if we called it `valueLogDensity`?

Reply:
Makes sense, will fix.

Reply:
I see some `Real.sum(...)`s in tests, do we want a `valueLogDensities`, too?